### PR TITLE
pkg/cloudprovider - fix golint failures

### PR DESCRIPTION
**Pull request by saravanan30erd**
**What this PR does / why we need it**:
Fixes golint issues - pkg/cloudprovider

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes #68026 

**Special notes for your reviewer**:

**Release note**:

```release-note
NONE
```

### CODE REVIEW

**Comment by saravanan30erd** [comment]
@neolit123 This PR is follow up of PR #68180 as I messed up there at rebase.

**Comment by neolit123** [comment]
thanks
/ok-to-test

**Comment by saravanan30erd** [comment]
@neolit123 The tests are failing due to below errors,

```
W0904 10:55:40.186] GoCompile: error running subcommand: exit status 1
I0904 10:55:40.286] /bazel-scratch/.cache/bazel/_bazel_root/e9f728bbd90b3fba632eb31b20e1dacd/sandbox/linux-sandbox/2098/execroot/__main__/pkg/controller/util/node/controller_utils.go:182:13: undefined: cloudprovider.InstanceNotFound
I0904 10:55:40.287] /bazel-scratch/.cache/bazel/_bazel_root/e9f728bbd90b3fba632eb31b20e1dacd/sandbox/linux-sandbox/2098/execroot/__main__/pkg/controller/util/node/controller_utils.go:198:12: undefined: cloudprovider.NotImplemented
```

I changed the var `cloudprovider.NotImplemented` to `cloudprovider.ErrNotImplemented` as per golint, please advice how to proceed.

File: CHANGELOG-1.10.md [hunk 1]
```diff
@@ -725,7 +725,7 @@ filename | sha256 hash
 * Add ipset and udevadm to the hyperkube base image. ([#61357](https://github.com/kubernetes/kubernetes/pull/61357), [@rphillips](https://github.com/rphillips))
 * kubectl: restore the ability to show resource kinds when displaying multiple objects ([#61985](https://github.com/kubernetes/kubernetes/pull/61985), [@liggitt](https://github.com/liggitt))
 * Remove 'system' prefix from Metadata Agent rbac configuration ([#61394](https://github.com/kubernetes/kubernetes/pull/61394), [@kawych](https://github.com/kawych))
-* Ensure cloudprovider.InstanceNotFound is reported when the VM is not found on Azure ([#61531](https://github.com/kubernetes/kubernetes/pull/61531), [@feiskyer](https://github.com/feiskyer))
+* Ensure cloudprovider.ErrInstanceNotFound is reported when the VM is not found on Azure ([#61531](https://github.com/kubernetes/kubernetes/pull/61531), [@feiskyer](https://github.com/feiskyer))
```
Comment (neolit123) [inline CHANGELOG-1.10.md]: i don't think we should update the old changelogs even if there were lint errors in the old code.
[1]

File: CHANGELOG-1.11.md [hunk 2]
```diff
@@ -864,7 +864,7 @@ You can now bind tokens to service requests. ([ref](https://github.com/kubernete
 * kubectl no longer renders a List as suffix kind name for CRD resources ([#62512](https://github.com/kubernetes/kubernetes/pull/62512), [@dixudx](https://github.com/dixudx))
 * Restored old behavior to the `--template` flag in `get.go`. In old releases, providing a `--template` flag value and no `--output` value implicitly assigned a default value ("go-template") to `--output`, printing using the provided template argument.
 ([#65377](https://github.com/kubernetes/kubernetes/pull/65377),[@juanvallejo](https://github.com/juanvallejo))
-* Ensured cloudprovider.InstanceNotFound is reported when the VM is not found on Azure ([#61531](https://github.com/kubernetes/kubernetes/pull/61531), [@feiskyer](https://github.com/feiskyer))
+* Ensured cloudprovider.ErrInstanceNotFound is reported when the VM is not found on Azure ([#61531](https://github.com/kubernetes/kubernetes/pull/61531), [@feiskyer](https://github.com/feiskyer))
```
Comment (neolit123) [inline CHANGELOG-1.11.md]: [1]

File: CHANGELOG-1.11.md [hunk 3]
```diff
@@ -1883,7 +1883,7 @@ filename | sha256 hash
 * Use inline func to ensure unlock is executed ([#61644](https://github.com/kubernetes/kubernetes/pull/61644), [@resouer](https://github.com/resouer))
 * `kubectl apply view/edit-last-applied support completion. ([#60499](https://github.com/kubernetes/kubernetes/pull/60499), [@superbrothers](https://github.com/superbrothers))
 * Automatically add system critical priority classes at cluster boostrapping. ([#60519](https://github.com/kubernetes/kubernetes/pull/60519), [@bsalamat](https://github.com/bsalamat))
-* Ensure cloudprovider.InstanceNotFound is reported when the VM is not found on Azure ([#61531](https://github.com/kubernetes/kubernetes/pull/61531), [@feiskyer](https://github.com/feiskyer))
+* Ensure cloudprovider.ErrInstanceNotFound is reported when the VM is not found on Azure ([#61531](https://github.com/kubernetes/kubernetes/pull/61531), [@feiskyer](https://github.com/feiskyer))
```
Comment (neolit123) [inline CHANGELOG-1.11.md]: [1]

File: CHANGELOG-1.9.md [hunk 4]
```diff
@@ -498,7 +498,7 @@ filename | sha256 hash
 * Backport Cloud CIDR allocator fixes to 1.9 ([#61797](https://github.com/kubernetes/kubernetes/pull/61797), [@satyasm](https://github.com/satyasm))
 * openstack cinder detach problem is fixed if nova is shutdowned ([#56846](https://github.com/kubernetes/kubernetes/pull/56846), [@zetaab](https://github.com/zetaab))
 * Fixed kube-proxy to work correctly with iptables 1.6.2 and later. ([#61734](https://github.com/kubernetes/kubernetes/pull/61734), [@danwinship](https://github.com/danwinship))
-* Ensure cloudprovider.InstanceNotFound is reported when the VM is not found on Azure ([#61531](https://github.com/kubernetes/kubernetes/pull/61531), [@feiskyer](https://github.com/feiskyer))
+* Ensure cloudprovider.ErrInstanceNotFound is reported when the VM is not found on Azure ([#61531](https://github.com/kubernetes/kubernetes/pull/61531), [@feiskyer](https://github.com/feiskyer))
```
Comment (neolit123) [inline CHANGELOG-1.9.md]: [1]

**Review by neolit123 — COMMENTED** [review]
@saravanan30erd thank you for the update!

i've added a couple of comments about updating the old change logs. other than that LGTM, but i would leave it to the maintainers to comment further.

FYI, k8s is currently in code freeze and this PR might not be merged until the code freeze is lifted towards the end of September.

File: CHANGELOG-1.10.md (hunk 1 above)
Comment (saravanan30erd) [reply CHANGELOG-1.10.md]: @neolit123 thanks. I reverted the changes in CHANGELOG.

**Comment by shubheksha** [comment]
@saravanan30erd, please use Ref #68026 instead of Fixes #68026 to avoid closing that issue

**Comment by saravanan30erd** [comment]
/assign @vishh @lavalamp 
please look on this

**Pull request closed without merging**